### ISSUE TITLE
Working, but not transacting.

### ISSUE THREAD

**Issue by derrickr** [arcbtc/Ziggy#1]
Hi,

I built the Lightning NF Pos, following your (most excellent) tutorial, but have run into problems with the transaction going to/from Opennode.

@ 12:35 in your youtube tutorial https://youtu.be/5A0KK0k33cI?t=755 you state: "in order to access the API keys you'll need to sign up for a dev account...  which is really simple to do in opennode...".

I therefore created a customer and merchant account, but as you mentioned as Dev accounts using: https://dev.opennode.com (as opposed to their main https://app.opennode.com/).

I then proceeded with the project and was successfully able to write to the NFC card, then enter an amount of Sats on the keypad which correctly shows on the display and when I present the card it successfully states: "Paid!".

I was thinking/hoping that being a test account that it wouldn't be using 'real' BTC transactions, so was suprised to see no transactions whatsoever in my opennode accounts.

I then thought that I needed to fund my customer account (kinda obvious?) but wasn't sure from where from, but again thought that since this is a dev/test scenario that I should perhaps try funding from BTC testnet. I therefore went ahead with this and successfully managed to fund my opennode customer dev account with 0.001 BTC from testnet.

Great! Try again, still nothing.

I'm now thinking that perhaps my above thought process has been too naive and that I should have funded with real BTC. However, I now can't remove/transfer the previous testnet 0.001BTC and I'm unsure what would happen if I were to now transfer some real BTC (i.e. non-testnet) into my opennode account if it would get mixed up with the testnet BTC.

I therefore went ahead with trying to create another customer account, but this time in https://app.opennode.com/ but have now come up against their KYC, which seems a lot of bother just to test.

Whilst not strictly technical, I'd appreciate any pointers so I can move forward.

Thanks!

Edit:
I've just been in touch with the opennode guys and have managed to trasnfer out the testnet BTC, so my 'dev' account is back to zero.

Not sure where to go from here. Should I fund this 'dev' account, or should I open a normal account in https://app.opennode.com/ and fund with real BTC?

**Comment by derrickr** [comment]
Update: I've been chatting with Kevin from opennode & upon his recommendation created a new account on app.opennode.com. Funded with a few quid's worth of real BTC & tried again, no luck.
Kevin informed me that "lightning invoices can be requested and paid by the same person" and I've created the relevant API keys to do so, but my Ziggy now returns with "error".

I'm looking into alternative methods for the Merchant to receive the transaction, thinking maybe Zap might work??

Would be great to get you thoughts/fixes.

Thanks.

**Comment by derrickr** [comment]
At last! I worked it out. There are 2 problems:

1. me being a newbie and not refreshing the opennode page to see any updated transactions.

But more importantly:

2. My ISP: VirginMedia. After searching for help & reading many, many articles, I was at my witts end, but then had the eureka moment: What if the network wasn't allowing the required protocols? I had a similar problem with my Raspiblitz, whereby I'd opened the required ports on my router but it still wasn't giving green indicative text, but changing to use Tor solved that.

I therefore used my phone as mobile wifi hotspot, restarted, Ziggy tethered and voila, transactions are flowing.

No more "error!", now I get "Paid!"

I tested again via my ISP, but it went back to "error!", but this time I at least knew to refresh the page and can see the transactions successfully coming in & out.

Learned a lot!
